### PR TITLE
Search class

### PR DESCRIPTION
**Pull request by anglofrog**
need help with concurrency issue in allimpl.go populate classmembers
AddCodeClass method seems to be way off of its target functionality

### CODE REVIEW

File: allimpl.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"fmt"
+
+	openapi "github.com/acceleratedlife/backend/go"
+	bolt "go.etcd.io/bbolt"
+)
+
+func UserByIdTx(tx *bolt.Tx, userId string) (user *bolt.Bucket, err error) {
```
Comment (saxonewt) [inline allimpl.go:10]: need to remove the function

File: staff.go [hunk 2]
```diff
@@ -23,17 +25,57 @@ func (s StaffApiServiceImpl) DeleteAuction(ctx context.Context, s2 string) (open
...
-func (s StaffApiServiceImpl) Deleteclass(ctx context.Context, s2 string) (openapi.ImplResponse, error) {
+func (s *StaffApiServiceImpl) Deleteclass(ctx context.Context, query openapi.RequestUser) (openapi.ImplResponse, error) {
 	//TODO implement me
 	panic("implement me")
 }
 
-func (s StaffApiServiceImpl) EditClass(ctx context.Context, body openapi.ClassesClassBody) (openapi.ImplResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (a *StaffApiServiceImpl) EditClass(ctx context.Context, body openapi.RequestEditClass) (openapi.ImplResponse, error) {
```
Comment (saxonewt) [inline staff.go:33]: move all staff functions to another file

File: allimpl.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		if string([]byte(k)) == userId {
+			println(k, v)
+		}
+
+	}
+	return nil, fmt.Errorf("school not found")
+
+}
+
+func ClassForAll(tx *bolt.Tx, classId string) (classBucket *bolt.Bucket, err error) {
```
Comment (saxonewt) [inline allimpl.go:29]: do you use this function anywhere?

Comment (chad187) [reply allimpl.go:29]: negative

File: staff.go (hunk 2 above)
Comment (chad187) [reply staff.go:33]: I don't understand what to move where?
If staff functions are not in staff.go then where should I put them?

Comment (anglofrog) [reply staff.go:33]: I don't understand what to move where?
If staff functions are not in staff.go then where should I put them?

Comment (saxonewt) [reply staff.go:33]: > I don't understand what to move where? If staff functions are not in staff.go then where should I put them?

staffimpl.go?

**Pull request merged**